### ISSUE TITLE
Curious about HRM on NLP tasks.

### ISSUE THREAD

**Issue by johnnyZeppelin** [sapientinc/HRM#61]
H and L will cooperate to try their best to make answers exactly the same as the labels in the dataset if the original loss function is kept. I tried this on wikitext-2-v1, using the same evaluation metrics. I found that after just a few epochs of training, the model can achieve an accuracy of over 50% on the test set.

If I'm understanding the model and metrics correctly, it seems that HRM can efficiently learn to answer purely textual questions exactly as they appear in the dataset. I'm curious if there's an upper limit to this performance. Will the model only eventually overfits on these purely text-based tasks? I think this is worth more investigating.

**Comment by Jamie-Stirling** [comment]
Hi, how are you structuring the input/output? Are you aligning sequence dimension to the HRM time dimension, or predicting in blocks? Curious to see your approach here.

**Comment by Sumching** [comment]
Hi! I noticed that your code feeds the model identical sequences for input and output. This encourages the network to memorize the identity mapping and overfit. During evaluation, do you mask or replace a portion of the tokens with a special symbol? If so, can the model still reach roughly 50 % accuracy?

**Comment by johnnyZeppelin** [comment]
Yes, you're right. I just notice that and I didn't do any more operations during evaluation. I think everyone who sees this discussion can try the masking or replacing as you suggests.  > Hi! I noticed that your code feeds the model identical sequences for input and output. This encourages the network to memorize the identity mapping and overfit. During evaluation, do you mask or replace a portion of the tokens with a special symbol? If so, can the model still reach roughly 50 % accuracy?
